File: movemork/Features/Paywall/PaywallReason.swift
//
//  PaywallReason.swift
//  movemork
//
//  MoveMark — Reusable paywall entry reasons.
//

import Foundation

enum PaywallReason: Equatable {
    case extraProperty
    case unlimitedExports
    case disputePacket
    case moveOutExport

    /// Primary paywall headline (shared across entry points).
    var headline: String {
        switch self {
        case .extraProperty:
            return "Protect more than one rental."
        case .unlimitedExports:
            return "Turn proof into more reports."
        case .disputePacket:
            return "Build a dispute-ready packet."
        case .moveOutExport:
            return "Document move-out proof."
        }
    }

    /// Context line under the shared Pro value prop.
    var subheadline: String {
        switch self {
        case .extraProperty:
            return "Free includes 1 proof vault. Add another when you move."
        case .unlimitedExports:
            return "You’ve used your free move-in report on this account."
        case .disputePacket:
            return "Organize photos and docs when your deposit is questioned."
        case .moveOutExport:
            return "Re-capture rooms and export move-out PDFs with Pro."
        }
    }

    var ctaTitle: String {
        "Start Pro"
    }
}

File: movemark-android/app/src/main/java/com/surensureshkumar/movemark/features/subscription/PaywallReason.kt
package com.surensureshkumar.movemark.features.subscription

enum class PaywallReason {
    ExtraProperty,
    UnlimitedExports,
    DisputePacket,
    MoveOutExport,
    MoveOutProof,
    MoveOutReport,
    ;

    val headline: String
        get() = when (this) {
            ExtraProperty -> "Protect more than one rental."
            UnlimitedExports -> "Turn proof into more reports."
            DisputePacket -> "Build a dispute-ready packet."
            MoveOutExport -> "Document move-out proof."
            MoveOutProof -> "Document move-out proof."
            MoveOutReport -> "Document move-out proof."
        }

    val subheadline: String
        get() = when (this) {
            ExtraProperty -> "Free includes 1 proof vault."
            UnlimitedExports -> "You've used your free move-in report on this account."
            DisputePacket -> "Organize photos and docs when your deposit is questioned."
            MoveOutExport -> "Re-capture rooms and export move-out reports with Pro."
            MoveOutProof -> "Re-capture each room before you return the keys."
            MoveOutReport -> "Make and share a move-out PDF from saved room proof."
        }

    val valueProp: String
        get() = com.surensureshkumar.movemark.core.growth.MoveMarkGrowthCopy.PAYWALL_VALUE_LINE
}
